addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -2,65 +2,61 @@ odoo.define('website_mass_mailing.editor', function (require) {
 'use strict';
 
 var core = require('web.core');
-var rpc = require('web.rpc');
+const Dialog = require('web.Dialog');
 var options = require('web_editor.snippets.options');
-var wUtils = require('website.utils');
 
 const qweb = core.qweb;
 var _t = core._t;
 
 
 options.registry.mailing_list_subscribe = options.Class.extend({
-    popup_template_id: "editor_new_mailing_list_subscribe_button",
-    popup_title: _t("Add a Newsletter Subscribe Button"),
-
-    //--------------------------------------------------------------------------
-    // Options
-    //--------------------------------------------------------------------------
-
     /**
-     * Allows to select mailing list.
-     *
-     * @see this.selectClass for parameters
+     * @override
      */
-    select_mailing_list: function (previewMode, value) {
-        var self = this;
-        var def = wUtils.prompt({
-            'id': this.popup_template_id,
-            'window_title': this.popup_title,
-            'select': _t("Newsletter"),
-            'init': function (field, dialog) {
-                return rpc.query({
-                    model: 'mailing.list',
-                    method: 'name_search',
-                    args: ['', [['is_public', '=', true]]],
-                    context: self.options.recordInfo.context,
-                }).then(function (data) {
-                    $(dialog).find('.btn-primary').prop('disabled', !data.length);
-                    var list_id = self.$target.attr("data-list-id");
-                    $(dialog).on('show.bs.modal', function () {
-                        if (list_id !== "0"){
-                            $(dialog).find('select').val(list_id);
-                        };
+    onBuilt() {
+        this._super(...arguments);
+        if (this.mailingLists.length) {
+            this.$target.attr("data-list-id", this.mailingLists[0][0]);
+        } else {
+            Dialog.confirm(this, _t("No mailing list found, do you want to create a new one? This will save all your changes, are you sure you want to proceed?"), {
+                confirm_callback: () => {
+                    this.trigger_up('request_save', {
+                        reload: false,
+                        onSuccess: () => window.location.href = '/web#action=mass_mailing.action_view_mass_mailing_lists',
```

Comment:
FWI, I kinda prefer to avoid the one line arrow function if the return is not meaningful (and could thus hide bugs or unintendend behaviors). The equivalent of what you wrote here is:
```javascript
function () {
    return window.location.href = '...';
}
```
-> if you wrote it that way, you would not have written the `return`.
So I would do:
```javascript
onSuccess: () => {
    window.location.href = '...';
},
```
Feel free to ignore or not :wink:

Reply:
It's not "wrong" for this specific code (could even make sense if you actually wanted to return the URL) :wink: Just a bad practice IMO